### PR TITLE
skip emit for properties named 'prototype'

### PR DESCRIPTION
**Pull request by evmar**
This would end up emitting Closure code like
  /** @type {string} */
  Foo.prototype.prototype;
which ends up confusing the compiler.

### CODE REVIEW

**Comment by mprobst** [comment]
@evmar have we inquired whether this is a fundamental limitation, or just a bug, or what?

**Comment by evmar** [comment]
I started a thread about it.  It's not clear exactly what's going wrong.  I think in terms of runtime behavior, no code actually wants to access this 'prototype' member, it's just doing it as a way of expressing a supertype constraint to TS, so I think it is harmless to swallow in this way.

File: src/jsdoc_transformer.ts [hunk 1]
```diff
@@ -339,6 +339,12 @@ function createClosurePropertyDeclaration(
         prop, `Skipping unnamed member:\n${escapeForComment(prop.getText())}`);
   }
 
+  if (name === 'prototype') {
+    // It's illegal to name a property 'prototype'.
```
Comment (mprobst) [inline src/jsdoc_transformer.ts]: Maybe expand this comment with what you wrote in the comment thread here?

Comment (evmar) [reply src/jsdoc_transformer.ts]: Done.

**Pull request merged**